### ISSUE TITLE
Shopify theme serve fails to sync all files of Dawn theme

### ISSUE THREAD

**Issue by aadviboutique** [Shopify/shopify-cli#1327]
# Issue summary

After cloning the dawn theme, when I am trying to run command "shopify theme serve", the commands fails citing errors in some of the files

## Expected behavior
Since I have not modified any files yet, the command should be able to sync the theme with shopify store and be able to host the theme

## Actual behavior

Its uploading partial set of files and as a result the I am getting an error when  I am trying to access the hosted theme locally or from the store

## Steps to reproduce the problem

1. Clone the Dawn theme using "shopify init"
1. Connect to store 
1. Run the command "shopify theme serve"

Gettting error:
<details>
* Syncing theme #124438970527 on  ***
┃                                                                                                                                                            100%
┃ ERROR update sections/apps.liquid:
┃   Liquid syntax error (line 3): Syntax error in tag 'render' - Template name must be a quoted string
┃ ERROR update sections/main-product.liquid:
┃   Liquid syntax error (line 74): Syntax error in tag 'render' - Template name must be a quoted string
┃ ERROR update templates/404.json:
┃   File extension is not valid for file templates/404.json
┃ ERROR update templates/article.json:
┃   File extension is not valid for file templates/article.json
┃ ERROR update templates/blog.json:
┃   File extension is not valid for file templates/blog.json
┃ ERROR update templates/cart.json:
┃   File extension is not valid for file templates/cart.json
┃ ERROR update templates/index.json:
┃   File extension is not valid for file templates/index.json
┃ ERROR update templates/collection.json:
┃   File extension is not valid for file templates/collection.json
┃ ERROR update templates/list-collections.json:
┃   File extension is not valid for file templates/list-collections.json
┃ ERROR update templates/page.contact.json:
┃   File extension is not valid for file templates/page.contact.json
┃ ERROR update templates/page.json:
┃   File extension is not valid for file templates/page.json
┃ ERROR update templates/password.json:
┃   File extension is not valid for file templates/password.json
┃ ERROR update templates/product.json:
┃   File extension is not valid for file templates/product.json
┃   Section type 'main-product' does not refer to an existing section file
┃ ERROR update templates/search.json:
┃   File extension is not valid for file templates/search.json
</details>

## Specifications

- App type: NA
- Operating System: Mac OS Big Sur (Ver 11.4)
- Shell: terminal
- Ruby version (ruby -v): ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]

**Comment by aadviboutique** [comment]
For example, this is the apps.liquid file, failing at line 3 [Syntax error in tag 'render' - Template name must be a quoted string ]

`<div class="{% if section.settings.include_margins %}page-width{% endif %}">`
 ` {%- for block in section.blocks -%}`
   **{% render block %}**
 ` {%- endfor -%}`
`</div>`

`{% schema %}
{
  "name": "t:sections.apps.name",
  "tag": "section",
  "class": "spaced-section",
  "settings": [
    {
      "type": "checkbox",
      "id": "include_margins",
      "default": true,
      "label": "t:sections.apps.settings.include_margins.label"
    }
  ],
  "blocks": [
    {
      "type": "@app"
    }
  ],
  "presets": [
    {
      "name": "t:sections.apps.presets.name"
    }
  ]
}
{% endschema %}
`

**Comment by waywardm** [comment]
![Capture2](https://user-images.githubusercontent.com/12498554/124146464-b2311a00-da85-11eb-84eb-472da9926ec7.PNG)


also shows this at the end too

**Comment by sickerman** [comment]
Been having the same issue with Shopify theme serve. any workarounds yet ?
<img width="827" alt="Screen Shot 2021-06-29 at 9 18 14 PM" src="https://user-images.githubusercontent.com/53154228/124329942-68feca00-db5a-11eb-8dbd-debeac7539ef.png">

**Comment by sickerman** [comment]
Heres how I resolved this. 

Online store 2.0 features and themes(DAWN) work with  new stores with developer preview. Even if you are the store owner you should add a staff account to your new development store with developer preview enabled and give access to Themes(Online store).  

When you receive the invite link create acc and login to your store admin. 

-`Shopify logout` from terminal.
-`Shopify login [URLTODEVSTORE]` eg `Shopify login johndoestore.myshopify.com` login in the browser with your staff account credentials not partner account.
- shopify theme init should work
- Shopify theme serve should sync with no issues. 

Let me know if this works. You can find more help in this [issue](https://github.com/Shopify/shopify-cli/issues/1309#issuecomment-871370263) which is now closed

**Comment by aadviboutique** [comment]
It did not work for me.. Not sure it worked for others or not.

**Comment by aadviboutique** [comment]
Seems like the issue is with liquid itself,

the render tag expects a template file, however in this we are passing a block variable
'{%- for block in section.blocks -%}'
'{% render block %}'
'{%- endfor -%}'

<img width="909" alt="Screenshot 2021-07-03 at 5 05 34 PM" src="https://user-images.githubusercontent.com/84778005/124352903-f0af0c00-dc20-11eb-8e34-ed64b9e8da8b.png">

<img width="973" alt="Screenshot 2021-07-03 at 5 07 37 PM" src="https://user-images.githubusercontent.com/84778005/124352962-38ce2e80-dc21-11eb-83fd-9d77067df861.png">

**Comment by sickerman** [comment]
Glad you figured it out.

**Comment by capatina** [comment]
Created a new store with developer preview enabled and still have this problem:
![image](https://user-images.githubusercontent.com/13295539/124630941-e410fa80-de50-11eb-9bae-e4f27c982c3a.png)

**Comment by aadviboutique** [comment]
@capatina are you running the command from outside of the theme folder? In that case you may try running "Shopify theme serve" from within the theme folder

**Comment by capatina** [comment]
just tried it with a totally new partner account, also doesn't work. Maybe one day in the far distant future things will just work when people say "it's simple"

**Comment by aadviboutique** [comment]
Also try installing wdm. "gem install wdm"

**Comment by capatina** [comment]
looks like there was another dawn folder inside the dawn folder. I deleted that one and it seems to work now and tried the sync, but it completed in under 1s and the updates are not registering. Im just getting more an more confused by this. Then i delete the dawn theme from the store and tried the sync again. Now the sync takes longer, but when i make changes to the files, the don't update. Also, the theme is not showing up in the list of themes on the store:

It's crazy how complicated this is getting

![image](https://user-images.githubusercontent.com/13295539/124643069-98fde400-de5e-11eb-8abc-1b488233d817.png)

**Comment by sickerman** [comment]
I had these same issues I was able to figure it out. By signing out of all shopify accounts and signing into dev store from terminal

**Comment by aadviboutique** [comment]
Yes... try to logout using "shopify logout". Close all browser sessions for shopify and the terminals. Open a new terminal and do a fresh login. Hope it works for you

**Comment by aadviboutique** [comment]
Also make sure that the account you are trying to sync dawn is a new partner account with the new developer preview. It will not work for older existing accounts ( I say this because I see the theme debut is there)
